contracts/actions/qidao/helpers/QiDaoRegistry.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import "../../../auth/AdminAuth.sol";
+
+contract QiDaoRegistry is AdminAuth {
+    error VaultAlreadyRegistered(uint16);
+
+    /// @notice IDs start from 1
+    mapping(address => uint16) public vaultIdByVaultAddress;
+    mapping(uint16 => address) public vaultAddressById;
```

Comment:
Maybe a check when we take vaultAddr as they could be 0x0, currently it isn't a problem as the code will just fail, but a low lvl. call to 0x0 might pass as success and leave some tokens on proxy etc...